src/core/public/application/types.ts

```diff
@@ -661,6 +661,28 @@ export interface InternalApplicationSetup extends Pick<ApplicationSetup, 'regist
   ): void;
 }
 
+/**
+ * Options for the {@link ApplicationStart.navigateToApp | navigateToApp API}
+ */
+export interface NavigateToAppOptions {
+  /**
+   * optional path inside application to deep link to.
+   * If undefined, will use {@link AppBase.defaultPath | the app's default path}` as default.
```

Comment:
Extracted the options type, as the inline definition was starting to be too big imho.